zipline/algorithm.py

```diff
@@ -845,6 +858,109 @@ def _create_daily_stats(self, perfs):
         )
         daily_stats = pd.DataFrame(daily_perfs, index=daily_dts)
 
+        sim_params = self.sim_params
+        data_portal = self.data_portal
+        perf_tracker = self.perf_tracker
+        asset_finder = data_portal.asset_finder
+        lookback_days = zp.DEFAULT_CVAR_LOOKBACK_DAYS
+
+        # Create a data frame of asset weights on each day of the simulation.
+        # If an asset was not held on a given date, it is assigned a weight of
+        # zero.
+        weights = pd.DataFrame(
+            perf_tracker.cumulative_risk_metrics.position_weights.tolist(),
+            index=daily_dts.normalize(),
+        ).fillna(0)
+
+        # For each day of the simulation, check to see if any futures were
+        # held. If so, convert it to a continuous future and assign its weight
+        # on that day to the continuous future. Once that is done, drop the
+        # futures contracts from the weights dataframe as we only want to look
+        # at the continuous futures when doing a history call later on.
+        futures_held = filter(
+            lambda asset: isinstance(asset, Future),
+            weights.columns,
+        )
+        for day in weights.index:
+            for asset in futures_held:
+                if weights.loc[day, asset] != 0:
+                    cf = zp.assets_for_history_call(asset, asset_finder, day)
```

Comment:
This could be costly; consider recording the continuous futres when recording the weights